packages/amplify-graphql-api-construct-tests/src/__tests__/sql-models.test.ts

```diff
@@ -0,0 +1,214 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import {
+  createNewProjectDir,
+  deleteDBInstance,
+  deleteDbConnectionConfig,
+  deleteProjectDir,
+  extractVpcConfigFromDbInstance,
+  setupRDSInstanceAndData,
+  storeDbConnectionConfig,
+} from 'amplify-category-api-e2e-core';
+import generator from 'generate-password';
+import { initCDKProject, cdkDeploy, cdkDestroy } from '../commands';
+import { graphql } from '../graphql-request';
+
+jest.setTimeout(1000 * 60 * 60 /* 1 hour */);
+
+interface DBDetails {
+  endpoint: string;
+  port: number;
+  dbName: string;
+  vpcConfig: {
+    vpcId: string;
+    securityGroupIds: string[];
+    subnetAvailabilityZones: {
+      subnetId: string;
+      availabilityZone: string;
+    }[];
+  };
+  ssmPaths: {
+    hostnameSsmPath: string;
+    portSsmPath: string;
+    usernameSsmPath: string;
+    passwordSsmPath: string;
+    databaseNameSsmPath: string;
+  };
+}
+
+describe('CDK GraphQL Transformer', () => {
+  let projRoot: string;
+  let projFolderName: string;
+
+  const [username, password, identifier] = generator.generateMultiple(3);
+
+  const region = process.env.AWS_REGION ?? 'us-east-1';
+
+  const dbname = 'default_db';
+
+  let dbDetails: DBDetails;
+
+  beforeAll(async () => {
+    dbDetails = await setupDatabase({
+      identifier,
+      engine: 'mysql',
+      dbname,
+      username,
+      password,
+      region,
+    });
+  });
+
+  afterAll(async () => {
+    await cleanupDatabase({ identifier: identifier, region, dbDetails });
+  });
+
+  beforeEach(async () => {
+    projFolderName = 'sqlmodels';
+    projRoot = await createNewProjectDir(projFolderName);
+  });
+
+  afterEach(async () => {
+    try {
+      const result = await cdkDestroy(projRoot, '--all');
+      console.log(`cdkDestroy result: ${JSON.stringify(result)}`);
+    } catch (_) {
+      /* No-op */
+    }
+
+    deleteProjectDir(projRoot);
+  });
+
+  ['2.80.0', 'latest'].forEach((cdkVersion) => {
+    test(`SQL Models base case - aws-cdk-lib@${cdkVersion}`, async () => {
+      const templatePath = path.resolve(path.join(__dirname, 'backends', 'sql-models'));
+      const name = await initCDKProject(projRoot, templatePath, cdkVersion);
+      writeDbDetails(dbDetails, projRoot);
+      const outputs = await cdkDeploy(projRoot, '--all');
+      const { awsAppsyncApiEndpoint: apiEndpoint, awsAppsyncApiKey: apiKey } = outputs[name];
+
+      const result = await graphql(
+        apiEndpoint,
+        apiKey,
+        /* GraphQL */ `
+          mutation CREATE_TODO {
+            createTodo(input: { description: "todo desc" }) {
+              id
+              description
+            }
+          }
+        `,
+      );
+
+      expect(result).toMatchSnapshot({
+        body: {
+          data: {
+            createTodo: {
+              id: expect.any(String),
+            },
+          },
+        },
+      });
```

Comment:
I know I was using snapshots in the `cdk-base` tests, but I think it's probably better to avoid snapshots, and just validate the data explicitly (e.g. response data and response codes) in the test itself.

Reply:
Fair enough. I'll make sure the expect assertions cover the data appropriately (which they should already)